wrapper/src/main/java/software/amazon/jdbc/plugin/AuroraConnectionDestroyerPlugin.java

```diff
@@ -0,0 +1,176 @@
+  private String instanceNameCol;
+  private String clusterInstanceTemplate;
+
+  AuroraConnectionDestroyerPlugin(PluginService pluginService, Properties props) {
+    this.pluginService = pluginService;
+    this.props = props;
+    this.rdsHelper = new RdsUtils();
+  }
+
+  @Override
+  public Set<String> getSubscribedMethods() {
+    return subscribedMethods;
+  }
+
+  @Override
+  public Connection connect(String driverProtocol, HostSpec hostSpec, Properties props,
+      boolean isInitialConnection, JdbcCallable<Connection, SQLException> connectFunc) throws SQLException {
+    if (driverProtocol.contains(PG_DRIVER_PROTOCOL)) {
+      this.retrieveInstanceQuery = PG_GET_INSTANCE_NAME_SQL;
+      this.instanceNameCol = PG_INSTANCE_NAME_COL;
+    } else {
+      this.retrieveInstanceQuery = MYSQL_GET_INSTANCE_NAME_SQL;
+      this.instanceNameCol = MYSQL_GET_INSTANCE_NAME_COL;
+    }
+
+    final Connection conn = connectFunc.call();
+
+    if (conn != null) {
+      final HostSpec currentHostSpec = this.pluginService.getCurrentHostSpec();
+      final String host = currentHostSpec.getHost();
+      if (rdsHelper.isRdsClusterDns(currentHostSpec.getHost())) {
+        currentHostSpec.addAlias(getInstanceEndpoint(conn, host));
+      }
+    }
+
+    ConnectionInvalidator.populateOpenedConnectionQueue(hostSpec, conn);
+    ConnectionInvalidator.logOpenedConnections();
+
+    return super.connect(driverProtocol, hostSpec, props, isInitialConnection, connectFunc);
+  }
+
+  @Override
+  public void initHostProvider(String driverProtocol, String initialUrl, Properties props,
+      HostListProviderService hostListProviderService,
+      JdbcCallable<Void, SQLException> initHostProviderFunc) throws SQLException {
+    if (hostListProviderService.isStaticHostListProvider()) {
+      hostListProviderService.setHostListProvider(new AuroraHostListProvider(
+          driverProtocol,
+          hostListProviderService,
+          props,
+          initialUrl));
+    }
+  }
+
+  private String getInstanceEndpointPattern(final String url) {
+    if (StringUtils.isNullOrEmpty(this.clusterInstanceTemplate)) {
+      this.clusterInstanceTemplate = AuroraHostListProvider.CLUSTER_INSTANCE_HOST_PATTERN.getString(this.props) == null
+          ? rdsHelper.getRdsInstanceHostPattern(url)
+          : AuroraHostListProvider.CLUSTER_INSTANCE_HOST_PATTERN.getString(this.props);
+    }
+
+    return this.clusterInstanceTemplate;
+  }
+
+  @Override
+  public <T, E extends Exception> T execute(Class<T> resultClass, Class<E> exceptionClass,
+      Object methodInvokeOn, String methodName, JdbcCallable<T, E> jdbcMethodFunc,
+      Object[] jdbcMethodArgs) throws E {
+
+    final HostSpec originalHost = this.pluginService.getCurrentHostSpec();
+
+    try {
+      T result = jdbcMethodFunc.call();
+      if ((methodName.contains(METHOD_CLOSE) || methodName.contains(METHOD_ABORT))) {
+        ConnectionInvalidator.invalidateAllConnections(originalHost);
```

Comment:
You probably need to remove the current connection from all queues, no to close all opened connections to the same node.

Reply:
That makes sense